File: Python/Easy/94.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        # inorder traversal travel left current then right
        # base case
        if root is None:
            return []
        answer = []

        # traverse the left side
        answer.extend(self.inorderTraversal(root.left))

        # add current value
        answer.append(root.val)

        # traverse the right side
        answer.extend(self.inorderTraversal(root.right))

        return answer

File: Java/Easy/94.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        // base case empty tree
        if (root == null) {
            return new ArrayList<>();
        }

        // inorder means go left then current then right
        List<Integer> answer = new ArrayList<>();

        // left
        answer.addAll(inorderTraversal(root.left));

        // current
        answer.add(root.val);

        // right
        answer.addAll(inorderTraversal(root.right));

        return answer;
    }
}